docs/processes/software_package_approval.md

```diff
@@ -0,0 +1,81 @@
+- privilege escalation attacks (enabling a user to gain elevated access permissions).
+
+(package_inclusion_policy)=
+
+## Policy
+
+- For each supported programming language, three package lists will be maintained:
+  - a core allowlist of broadly useful packages that should be pre-installed in each environment
+  - an extra allowlist of packages that may be useful for specific projects
+  - an expanded list to be made available from the package repositories consisting of the core and extra packages plus their dependencies
+- Users may request to add packages to these allowlists via the {ref}`package request procedure <package_request_procedure>`.
+  - In the interests of improving researcher productivity the aim will be to accommodate such requests, provided there are no outweighing security concerns associated with the package or its dependencies.
+- Requests will be reviewed by the project team. The reviewer may take into account:
+  - information provided by the user when making the request
+  - package author/contributor identities
+  - any existing package/version blacklist
+  - relevant data on the package _and_ its full dependency tree including:
+    - download statistics (recent and longer-term, current and previous versions)
+    - publicly-accessible CVE databases (listing Common Vulnerabilities and Exposures)
+- If approved, a requested package will be added to either the core or extra allowlist (as appropriate)
+
+(package_inclusion_criteria)=
+
+### Criteria for inclusion in core
+
+Only software that is considered broadly useful to a cross section of researchers should be included in core.
+
+To meet this condition, a package should:
+
+- implement at least one generic (i.e. not domain-specific) statistical algorithm or method, or
+- provide support for a cross-cutting analysis technique (e.g. geospatial data analysis, NLP), or
+- facilitate data science or software development best practices (e.g. for robustness, correctness, reproducibility), or
+- enhance the presentational features of the programming language (e.g. for producing plots, notebooks, articles, websites), or
+- enhance the usability of the programming language or development environment (e.g. RStudio, PyCharm).
+
+(package_request_procedure)=
+
+## Package request/review procedure
+
+1. A user requests a package by opening a `Software package request` issue on the Data Safe Haven GitHub repository, including responses to the following questions:
+
+- Is this package the mostly widely supported for the intended purpose?
+- What will you be able to do with this package that you can't currently do? What alternatives are there?
+- What risks to data integrity/security might arise from including this package or its dependencies?
+
+2. A member of the project team reviews the request according to the terms of the {ref}`package_inclusion_policy`.
+
+3. The reviewer adds their decision (accept/reject) to the issue and notifies the user who made the request.
```

Comment:
If I was a reviewer, how would I know how to make this decision? Is it possible to expand some guidance here - for example, in Daniel from Edon's case, would the reviewer just google "R Arrow" and check it's a real package and somehow figure out if it's something without known security vulnerabilities?

Reply:
The issue template asks them to answer the following questions:

- What will you be able to do with this package that you can't currently do?
- Is this the most widely supported package for the intended purpose? What alternatives are there?
- What risks to data integrity/security might arise from including this package or its dependencies?

Reply:
Makes sense, so the burden of checking risks falls to the requester

Reply:
The burden of explaining why the risks are manageable. We still don't have a well-defined way to make a decision based on these answer though :)